File: evclarity/app/src/main/java/com/stewartmcm/evclarity/TripAdapter.java
package com.stewartmcm.evclarity;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.stewartmcm.evclarity.db.PredictEvDatabaseHelper;
import com.stewartmcm.evclarity.model.Trip;

import java.text.DecimalFormat;
import java.util.ArrayList;

public class TripAdapter extends RecyclerView.Adapter<TripAdapter.TripAdapterViewHolder> {

    private static final String TAG = "TRIP_ADAPTER";
    private ArrayList<Trip> trips;

    public TripAdapter(Context context) {
        PredictEvDatabaseHelper helper = PredictEvDatabaseHelper.getInstance(context);
        trips = helper.getAllTrips();
    }

    @NonNull
    @Override
    public TripAdapterViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View item = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item_trip, parent, false);
        return new TripAdapterViewHolder(item);
    }

    @Override
    public void onBindViewHolder(TripAdapterViewHolder holder, int position) {
        float miles = trips.get(position).getMiles();
        float savings = trips.get(position).getSavings();

        DecimalFormat milesFormat = new DecimalFormat("##0.0");
        DecimalFormat savingsFormat = new DecimalFormat("###.00");

        String savingsString = savingsFormat.format(savings);
        Log.i(TAG, "savings: " + savingsString);

        String milesString = milesFormat.format(miles);
        Log.i(TAG, "mileage: " + milesString);

        holder.mileageTextView.setText(milesString + " miles");
        holder.savingsTextView.setText("$" + savingsString);
        holder.dateTimeTextView.setText(trips.get(position).getTimeStamp());
    }

    @Override
    public int getItemCount() {
        return trips.size();
    }

    static class TripAdapterViewHolder extends RecyclerView.ViewHolder {
        TextView mileageTextView;
        TextView dateTimeTextView;
        TextView savingsTextView;
        TripAdapterViewHolder(View itemView) {
            super(itemView);
            mileageTextView = (TextView) itemView.findViewById(R.id.list_item_mileage);
            dateTimeTextView = (TextView) itemView.findViewById(R.id.list_item_date);
            savingsTextView = (TextView) itemView.findViewById(R.id.list_item_savings);
        }

    }

    public int removeTrip(int tripPosition) {
        trips.remove(tripPosition);
        return trips.size();
    }

}

File: evclarity/app/src/main/java/com/stewartmcm/evclarity/TripAdapter.kt
package com.stewartmcm.evclarity

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.stewartmcm.evclarity.db.PredictEvDatabaseHelper
import com.stewartmcm.evclarity.model.Trip
import java.text.DecimalFormat
import java.util.*

class TripAdapter(context: Context) : RecyclerView.Adapter<TripAdapter.TripViewHolder>() {
    private val trips: ArrayList<Trip>

    init {
        val helper = PredictEvDatabaseHelper(context)
        trips = helper.allTrips
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TripViewHolder {
        val item = LayoutInflater.from(parent.context).inflate(R.layout.list_item_trip, parent, false)
        return TripViewHolder(item)
    }

    override fun onBindViewHolder(holder: TripViewHolder, position: Int) {
        val miles = trips[position].miles
        val savings = trips[position].savings

        val milesFormat = DecimalFormat("##0.0")
        val savingsFormat = DecimalFormat("###.00")

        val savingsString = savingsFormat.format(savings.toDouble())
        Log.i(TAG, "savings: $savingsString")

        val milesString = milesFormat.format(miles.toDouble())
        Log.i(TAG, "mileage: $milesString")

        holder.mileageTextView.text = "$milesString miles"
        holder.savingsTextView.text = "$$savingsString"
        holder.dateTimeTextView.text = trips[position].timeStamp
    }

    override fun getItemCount(): Int {
        return trips.size
    }

    class TripViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var mileageTextView: TextView
        var dateTimeTextView: TextView
        var savingsTextView: TextView

        init {
            mileageTextView = itemView.findViewById<View>(R.id.list_item_mileage) as TextView
            dateTimeTextView = itemView.findViewById<View>(R.id.list_item_date) as TextView
            savingsTextView = itemView.findViewById<View>(R.id.list_item_savings) as TextView
        }

    }

    fun removeTrip(tripPosition: Int): Int {
        trips.removeAt(tripPosition)
        return trips.size
    }

    companion object {
        private val TAG = "TRIP_ADAPTER"
    }

}